### ISSUE TITLE
SIGSEGV when running queries

### ISSUE THREAD

**Issue by jacobobryant** [xtdb/xtdb#1552]
Sometimes when running queries, RocksDB throws SIGSEGV. Usually (but not always) it happens when the query is inside a `pmap` or `future`. This happens both on my local machine and on a digital ocean vps. When I do run into it, it reproduces each time I run the same query. So far I've just been removing my calls to `pmap`/`future` which usually fixes it (too lazy to bother :) ), but I'll try to put together a minimal example and post that here today/monday.

(I'm on 21.06-1.17.1-beta)

**Comment by refset** [comment]
Are you using either `open-db` or `open-q` here? That could explain things, since they're not designed with thread-safety in mind.

**Comment by jarohen** [comment]
As @refset says, both `open-db` and `open-q` take out resources that are only intended to be used from a single thread. I haven't tried it myself (YMMV :smile:), but I'd be tempted to try wrapping the result in `seque` if you want to continue processing the result in a `pmap`, to ensure that the results themselves are only generated on a single thread. 

Regarding the issue once you bring it back to a single thread, the only time I'm aware we've seen this thus far is the usual Clojure laziness+with-open issue, where the lazy-seq escapes the bounds of the `with-open` - assuming this isn't the case, I'm interested to see the repro!

James

**Comment by jacobobryant** [comment]
Tried to respond to this by email the other day, looks like it didn't go through. anyway, yes! I am indeed using open-db/open-q often. I realized after the fact that the most recent occurrence of this was most likely due to using the open-db result after it was closed, though that hasn't always been the case when I've run into this. I'll retry some of the other instances with crux.api/db + wrapping in seque and hopefully will end up just closing this :).

**Comment by jacobobryant** [comment]
OK, I was able to find and reproduce one previous instance of this, and that was solved by not using open-db (it involved doing some regular queries inside a pmap). I'm guessing that's been the cause of most/all the times I've hit this (apart from the using-db-after-it's-been-closed thing)--I didn't realize before opening this issue that the query resources weren't thread safe. I'll be sure to try `seque` if I run into this again while consuming from `open-q`. Thanks!
